src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -240,6 +250,50 @@ private void transferRoutes(){
                     e.getIn().setBody(objectMapper.writeValueAsString(response));
                 });
 
+        //fetch transfer details based on client correaltion id
+        from("rest:GET:/channel/txnState/{correlationId}")
+                .id("transfer-details-correlationid")
+                .log(LoggingLevel.INFO, "## CHANNEL -> inbound transferDetail request for client correlation Id" +
+                        " ${header.correlationId}")
+                .process(e -> {
+                    String tenantId = e.getIn().getHeader("Platform-TenantId", String.class);
```

Comment:
can you reduce the number of lines in this block by moving identical lines into a method?

Reply:
I have modularised the code further, if needed I will try to modularise more